### ISSUE TITLE
tensorflow.python.keras.applications.vgg16 model got low accuracy on the ILSVRC2012 validation dataset

### ISSUE THREAD

**Issue by nicolaswilde** [tensorflow/tensorflow#50150]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1.  It must be a bug, a feature request, or a significant problem with the
    documentation (for small docs fixes please send a PR instead).
2.  The form below must be filled out.
3.  It shouldn't be a TensorBoard issue. Those go
    [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: No
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: NA
-   **TensorFlow installed from (source or binary)**: pip install
-   **TensorFlow version (use command below)**: 2.4
-   **Python version**: 3.7
-   **Bazel version (if compiling from source)**: NA
-   **GCC/Compiler version (if compiling from source)**: NA
-   **CUDA/cuDNN version**: CUDA 11.0/cuDNN 8.0
-   **GPU model and memory**: GTX1060 6G
-   **Exact command to reproduce**: NA

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
I imported the VGG-16 model from tensorflow.python.keras.applications.vgg16 and evaluated it on the ILSVRC2012 validation dataset. According to "https://keras.io/api/applications/", the top-1 and top-5 accuracy of VGG-16 model should be 71.3% and 90.1%, but I got only 65.7% and 86.8% on the ILSVRC2012 validation dataset.

I downloaded the validation dataset and loaded it using "image_dataset_from_directory()" imported from tensorflow.keras.preprocessing. I preprocessed the images using "preprocess_input()" imported from tensorflow.python.keras.applications.vgg16. I have also dealed with the inconsistency between the original imagenet ILSVRC2012_ID and the class index which the pre-trained model uses.

I wonder why I got low accuracy. I suspect that the image resize interpolation method affects data destribution. I suggest that the image preprocessing method should be explicitly listed since it might vary from model to model.

I also found a bug. The function image_dataset_from_directory( ..., interpolation='nearest') will return a dataset of which the datat type is 'uint8' instead of 'float32', while other interpolation methods always return 'float32' dataset.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

source code:
```
from __future__ import print_function as _print_function

import sys as _sys

from tensorflow.python.keras.applications.vgg16 import VGG16
from tensorflow.python.keras.applications.vgg16 import decode_predictions
from tensorflow.python.keras.applications.vgg16 import preprocess_input
from tensorflow.keras.preprocessing import image_dataset_from_directory

import numpy as np

def preprocess(img, label):
    img = preprocess_input(img)
    return img, label

if __name__ == '__main__':
    model = VGG16(include_top=True, weights='imagenet')
    #model.compile(optimizer="adam",loss='sparse_categorical_crossentropy',metrics=['accuracy']) # 65.68%
    model.compile(optimizer="adam",loss='sparse_categorical_crossentropy',metrics=['sparse_top_k_categorical_accuracy']) # 86.75%

    ds = image_dataset_from_directory('C:\ILSVRC2012\ILSVRC2012_img_val',
        labels=list(np.loadtxt('new_val_truth.txt', dtype='int')),
        label_mode='int', color_mode='rgb', batch_size=32,
        image_size=(224, 224), shuffle=False, interpolation='bilinear')
    ds = ds.map(preprocess)

    model.evaluate(ds, verbose=1)
```

logs:
```
1563/1563 [==============================] - 684s 432ms/step - loss: 1.4227 - accuracy: 0.6568
1563/1563 [==============================] - 682s 430ms/step - loss: 1.4227 - sparse_top_k_categorical_accuracy: 0.8675
```

**Comment by nicolaswilde** [comment]
@UsharaniPagadala 
Thanks a lot! I have found the problem! According to "https://stackoverflow.com/questions/58248297/keras-applications-vgg16-low-accuracy-on-imagenet", the 224x224x3 images should be created by a center crop. For example, a 500 * 375 image should be firstly cropped to a 375 * 375 image, then resized to 224 * 224.

I found that tensorflow.python.keras.preprocessing.image provides the function defined as "smart_resize(x, size, interpolation='bilinear')". Then I modified the regular resizing method `img = image_ops.resize_images_v2(img, image_size, method=interpolation)` in `tensorflow.python.keras.preprocessing.image_dataset.path_to_image(path, image_size, num_channels, interpolation):` by `img = image.smart_resize(img, image_size, interpolation=interpolation)`. When I used the "smart_resize" function to resize the images, I got 70.08%(interpolation='bicubic') top1-accuracy, which is close to 71.3% listed in the keras document. But I still have no idea how to improve the rest 1.2% accuracy.

During the experiment I found another bug (Actually I have no idea whether it is a bug). When setting verbose = 1, the dynamic accuracy log is not consistent with the real accuracy. For example, when I run the following code:
```
loss, acc1, acc5 = model.evaluate(ds, verbose=1)
print(acc1, acc5)
```
I got:
```
1563/1563 [==============================] - 772s 488ms/step - loss: 1.2287 - sparse_categorical_accuracy: 0.7007 - sparse_top_k_categorical_accuracy: 0.8932
0.6998400092124939 0.892520010471344
```
The accuracies is a little bit different.
